.github/workflows/yocto-build-deploy.yml

```diff
@@ -585,6 +595,9 @@ jobs:
 
           cat "${AUTO_CONF_FILE}"
 
+          >&2 eval "$(ssh-agent)"
+          echo "${{ secrets.YOCTO_SSH_PRIVATE_KEY_B64 }}" | base64 -d | ssh-add - >&2
```

Comment:
[PEM formatted private keys](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail) like those used by OpenSSH are already base64 encoded, with a header and footer wrapping the encoded data.

These labels can be stripped and the base64 content passed as the secret, and they can be reinserted before passing the key to `ssh-add`.

This mirrors the approach used with `SIGN_KMOD_KEY_APPEND`. https://github.com/balena-os/meta-balena/pull/3114/files

Reply:
The problem is the header and the footer and newlines are not suitable for storing as Git secrets unless we update them by hand. What is the best format to retain the secret as a single one-line string?

Reply:
@jakogut why is that preferred?